### PR TITLE
PDK template sync

### PR DESCRIPTION
**Pull request by DavidS**
These change bring the  modulesync_config templates and setup closer to the current pdk-templates. Merging these rather pedestrian changes will be a first step towards completely switching over to PDK and PDKsync.

Splitting the easy changes out like this, clears the path to investigating the remaining more complicated changes.

See https://github.com/DavidS/pdk-templates/tree/pdk-template-sync for more WIP on the other side.

For now, please pick apart the changes here, to make sure that I'm following your preferences, and haven't bolloxed up something. i've only tested this against a single module and have not even run tests or validation over that yet.

### CODE REVIEW

File: moduleroot/Rakefile.erb [hunk 1]
```diff
@@ -1,11 +1,13 @@
 require 'puppetlabs_spec_helper/rake_tasks'
+require 'puppet-syntax/tasks/puppet-syntax'
+require 'puppet_blacksmith/rake_tasks' if Bundler.rubygems.find_name('puppet-blacksmith').any?
+require 'github_changelog_generator/task' if Bundler.rubygems.find_name('github_changelog_generator').any?
+require 'puppet-strings/tasks' if Bundler.rubygems.find_name('puppet-strings').any?
```
Comment (ekohl) [inline moduleroot/Rakefile.erb]: IMHO this should be a part of puppetlabs_spec_helper: https://github.com/puppetlabs/puppetlabs_spec_helper/pull/274

File: moduleroot/Rakefile.erb [hunk 2]
```diff
@@ -1,11 +1,13 @@
 require 'puppetlabs_spec_helper/rake_tasks'
+require 'puppet-syntax/tasks/puppet-syntax'
```
Comment (ekohl) [inline moduleroot/Rakefile.erb]: Looks redundant: https://github.com/puppetlabs/puppetlabs_spec_helper/blob/8ccffc732dc26c741deaabd8aaeb71b3a1326f71/lib/puppetlabs_spec_helper/rake_tasks.rb#L200

File: moduleroot/spec/default_facts.yml.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Use default_module_facts.yml for module specific facts.
```
Comment (ekohl) [inline moduleroot/spec/default_facts.yml.erb]: We got rid of this and I'd prefer to not introduce it again: https://github.com/voxpupuli/modulesync_config/commit/04459bff77720540ee1ed7d12303e2cd9b72b585

File: moduleroot/spec/spec_helper.rb.erb [hunk 4]
```diff
@@ -11,12 +11,23 @@ require 'puppetlabs_spec_helper/module_spec_helper'
 require 'rspec-puppet-facts'
 include RspecPuppetFacts
 
-if File.exist?(File.join(__dir__, 'default_module_facts.yml'))
-  facts = YAML.load(File.read(File.join(__dir__, 'default_module_facts.yml')))
-  if facts
-    facts.each do |name, value|
-      add_custom_fact name.to_sym, value
-    end
+default_facts = {
```
Comment (ekohl) [inline moduleroot/spec/spec_helper.rb.erb]: I prefer the other style. We rely on rspec-puppet-facts and that takes care of this for us.

File: moduleroot/Rakefile.erb (hunk 1 above)
Comment (DavidS) [reply moduleroot/Rakefile.erb]: excellent point. I'll move it over

File: moduleroot/Rakefile.erb (hunk 2 above)
Comment (DavidS) [reply moduleroot/Rakefile.erb]: yup, it is.

File: moduleroot/spec/default_facts.yml.erb (hunk 3 above)
Comment (DavidS) [reply moduleroot/spec/default_facts.yml.erb]: Looking over it, all of this default_facts.yml is redundant anyways. I'm gonna nuke it from orbit.

File: moduleroot/spec/spec_helper.rb.erb (hunk 4 above)
Comment (DavidS) [reply moduleroot/spec/spec_helper.rb.erb]: see above. :nuke:

**Comment by ekohl** [comment]
@DavidS we've done some cleanups in master. Would you mind rebasing this?

File: moduleroot/Rakefile.erb (hunk 1 above)
Comment (DavidS) [reply moduleroot/Rakefile.erb]: This has now been released in puppetlabs_spec_helper 2.14.0

File: moduleroot/spec/default_facts.yml.erb (hunk 3 above)
Comment (DavidS) [reply moduleroot/spec/default_facts.yml.erb]: I've taken this commit out for now, as I didn't get around to fix this on the pdk-templates side.

File: moduleroot/spec/spec_helper.rb.erb (hunk 4 above)
Comment (DavidS) [reply moduleroot/spec/spec_helper.rb.erb]: I've taken this commit out for now, as I didn't get around to fix this on the pdk-templates side.

**Comment by DavidS** [comment]
This first force-push was only to work in ekohl's comments and make sure that the code as it was two weeks ago is fine. I'll do the full rebase next. wish me luck :-)

**Comment by DavidS** [comment]
This now contains the rebase on top of 2.7.0. As feared the compare page is a lot less helpful to understand the diff :-/

Unless I broke something or someone else finds fault in the changes here, I think this is good to go in as a first step.

**Comment by ekohl** [comment]
I toyed with a rebase to see if I could use our modulesync repo as a PDK template: https://github.com/ekohl/modulesync_config/commits/pdk-style. PDK barfed for me on bundler after correctly syncing.

Overall I'm still not convinced about PDK. I really hate that it puts the PDK version in `metadata.json` which IMHO only adds noise. It also downgrades various gems while pulling in a lot more than I'd have a need for. I won't hold back using PDK in Voxpupuli but I don't see myself using it any time soon.

That said, I still like the goal of a very minimal set of synced files. Reading the git log and only seeing noise in modulesynced files is very distracting.

File: moduleroot/Rakefile.erb [hunk 5]
```diff
@@ -71,8 +71,7 @@ task :reference, [:debug, :backtrace] do |t, args|
   Rake::Task['strings:generate:reference'].invoke(patterns, args[:debug], args[:backtrace])
 end
 
-begin
-  require 'github_changelog_generator/task'
+if Bundler.rubygems.find_name('github_changelog_generator').any?
```
Comment (ekohl) [inline moduleroot/Rakefile.erb]: How about `if defined? GitHubChangelogGenerator` so you don't rely on bundler?

File: moduleroot/spec/spec_helper.rb.erb [hunk 6]
```diff
@@ -34,33 +36,30 @@ puts 'augeasproviders: setting Puppet[:libdir] to work around broken type autolo
...
+  end
+  <%- end -%>
+  c.filter_run_excluding(bolt: true) unless ENV['GEM_BOLT']
   c.after(:suite) do
     RSpec::Puppet::Coverage.report!
   end
 end
-<%- [@configs['spec_overrides']].flatten.compact.each do |line| -%>
 
+def ensure_module_defined(module_name)
```
Comment (ekohl) [inline moduleroot/spec/spec_helper.rb.erb]: Should this live in puppetlabs_spec_helper instead?

**Comment by ekohl** [comment]
@DavidS any plans to continue this effort or should we close it?

**Comment by ekohl** [comment]
Given the lack of activity and that we've since introduced our own `voxpupuli-test` and `voxpupuli-acceptance` gems I'm closing this now.

**Pull request closed without merging**